todo-app/src/components/TodoItem.vue

```diff
@@ -0,0 +1,58 @@
+<template>
+  <div class="flex flex-row sm:flex-col justify-between items-center">
+    <div class="flex flex-row sm:justify-between sm:w-full">
+      <div class="font-semibold text-lg sm:text-4xl">
+        {{ props.todo.title }}
+      </div>
+      <div
+        class="w-2 h-2 ml-2 mt-3 sm:w-32 sm:h-10 rounded-full flex items-center justify-center"
+        :class="colorMap[props.todo.importance]"
+      >
+        <div class="hidden sm:block text-lg font-semibold text-white">
+          {{ props.todo.importance }}
+        </div>
+      </div>
+    </div>
+    <div
+      class="flex flex-col sm:flex-row justify-between sm:pt-6 sm:mt-6 sm:w-full"
+    >
+      <div class="text-neutral-500 hidden sm:block sm:text-2xl font-semibold">
+        {{ props.todo.content }}
+      </div>
+      <button class="relative" @click="handleCheckClick">
+        <GreenEllipseIcon class="w-6 sm:w-10" v-if="checked" />
+        <GreenCheckIcon
+          class="absolute bottom-1 sm:left-1 sm:bottom-2 w-6 sm:w-10"
+          v-if="checked"
+        />
+        <BlackEllipseIcon class="w-6 sm:w-10" v-if="!checked" />
+      </button>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from "vue";
+import { ColorMap } from "../types/IColorMap.vue";
+import GreenCheckIcon from "./icons/GreenCheckIcon.vue";
+import GreenEllipseIcon from "./icons/GreenEllipseIcon.vue";
+import BlackEllipseIcon from "./icons/BlackEllipseIcon.vue";
+import { Todo } from "../types/ITodoItem.vue";
+
+interface Props {
+  todo: Todo,
+  index: number,
+}
+
+const props = defineProps<Props>();
+const checked = ref<boolean>(false);
```

Comment:
If its a boolean always use is prefix, also for primitives you dont have to specifically type refs, typescript would recognise based on the initial value